test/asyncapiSchemaFormatParser_test.js

```diff
@@ -98,4 +109,38 @@ describe('asyncapiSchemaFormatParser', function() {
 
     expect(async () => await parser.parse(parsedInput)).to.not.throw();
   });
+
+  it('should deep clone schema into x-parser-original-payload', async function() {
+    const asyncapi = fs.readFileSync(path.resolve(__dirname, './good/asyncapi-complex-schema.yml'), 'utf8');
+    const expectedOriginalPayload = {
+      type: 'object',
+      properties: {
+        id: {
+          description: 'Id of the streetlight.',
+          minimum: 0,
+          type: 'integer'
+        },
+        lumens: {
+          description: 'Light intensity measured in lumens.',
+          minimum: 0,
+          type: 'integer'
+        },
+        sentAt: {
+          description: 'Date and time when the message was sent.',
+          format: 'date-time',
+          type: 'string'
+        }
+      }
+    };
+    const expectedOriginalSchemaFormat = 'application/vnd.aai.asyncapi;version=2.0.0';
+    
+    const parsedOutput = await parser.parse(asyncapi);
+    const parsedMessage = parsedOutput.channel('light/measured').publish().message().json();
+
+    expect(parsedMessage.payload).to.have.property('x-parser-schema-id');
+    expect(parsedMessage).to.have.property('x-parser-original-payload');
+    expect(parsedMessage).to.have.property('x-parser-original-schema-format');
+    expect(parsedMessage['x-parser-original-payload']).to.deep.equal(expectedOriginalPayload);
```

Comment:
This is the extra test I meant above:

```suggestion
    expect(parsedMessage['x-parser-original-payload']).to.deep.equal(expectedOriginalPayload);
    expect(parsedMessage['x-parser-original-payload']).to.not.equal(parsedMessage.payload);
```

We just want to make sure that `x-parser-original-payload` and `message.payload` are not both the same object.